spec/lib/carma/models/submission_spec.rb

```diff
@@ -0,0 +1,181 @@
+
+      expect(submission.metadata).to be_instance_of(described_class::Metadata)
+      expect(submission.metadata.claim_id).to eq(claim.id)
+    end
+
+    it 'will override :claim_id when passed in metadata and use claim.id instead' do
+      claim = build(:caregivers_assistance_claim)
+
+      submission = described_class.from_claim(claim, claim_id: 99)
+
+      expect(submission).to be_instance_of(described_class)
+      expect(submission.data).to eq(claim.parsed_form)
+      expect(submission.carma_case_id).to eq(nil)
+      expect(submission.submitted_at).to eq(nil)
+
+      expect(submission.metadata).to be_instance_of(described_class::Metadata)
+      expect(submission.metadata.claim_id).to eq(claim.id)
+    end
+  end
+
+  describe '#metadata=' do
+    it 'uses the Metadata class' do
+      submission = described_class.new
+      submission.metadata = { claim_id: 100 }
+      expect(submission.metadata).to be_instance_of(described_class::Metadata)
+      expect(submission.metadata.claim_id).to eq(100)
+    end
+
+    it 'returns true if :submitted_at is set' do
+      submission = described_class.new(submitted_at: DateTime.now.iso8601)
+      expect(submission.submitted?).to eq(true)
+    end
+
+    it 'returns false if :carma_case_id and :submitted_at are falsy' do
+      submission = described_class.new
+      expect(submission.submitted?).to eq(false)
+    end
+  end
+
+  describe '::request_payload_keys' do
+    it 'inherits fron Base' do
+      expect(described_class.ancestors).to include(CARMA::Models::Base)
+    end
+
+    it 'sets request_payload_keys' do
+      expect(described_class.request_payload_keys).to eq(%i[data metadata])
+    end
+  end
+
+  describe '#to_request_payload' do
+    it 'can receive :to_request_payload' do
+      metadata = described_class.new data: { my: 'data' }, metadata: { claim_id: 1234 }
+      expect(metadata.to_request_payload).to eq(
+        {
+          'data' => { my: 'data' },
+          'metadata' => { 'claimId' => 1234 }
+        }
+      )
+    end
+  end
+
+  describe '#submitted?' do
+    it 'returns true if :carma_case_id is set' do
+      submission = described_class.new(carma_case_id: 'aB935000000A9GoCAK')
+      expect(submission.submitted?).to eq(true)
+    end
+
+    it 'returns true if :submitted_at is set' do
+      submission = described_class.new(submitted_at: DateTime.now.iso8601)
+      expect(submission.submitted?).to eq(true)
+    end
+
+    it 'returns false if :carma_case_id and :submitted_at are falsy' do
+      submission = described_class.new
+      expect(submission.submitted?).to eq(false)
+    end
+  end
+
+  describe '#submit!' do
+    let(:submission) do
+      CARMA::Models::Submission.from_claim(
+        build(:caregivers_assistance_claim)
+      )
+    end
+
+    context 'when :data is invalid' do
+      xit 'raises exception' do
+      end
+    end
+
+    context 'when :metadata is invalid' do
+      xit 'raises exception' do
+      end
+    end
```

Comment:
Need to be implemented?

Reply:
Removed. I decided not to add `ActiveRecord::Validations` with a `:valid?` pattern and just build form a single `:from_claim` method that properly assigns values. I may use `ActiveRecord::Validations` in the future if consumers of this library (aka `app`) requires a more atomic interface.